src/app/libp2p_helper/src/libp2p_helper/main.go

```diff
@@ -306,16 +362,21 @@ func (s *subscribeMsg) run(app *app) (interface{}, error) {
 		for {
 			msg, err := sub.Next(ctx)
 			if err == nil {
-				data := b58.Encode(msg.Data)
-				app.writeMsg(publishUpcall{
-					Upcall:       "publish",
-					Subscription: s.Subscription,
-					Data:         data,
-					Sender:       peer.IDB58Encode(msg.ReceivedFrom),
-				})
+				sender, err := findPeerInfo(app, msg.ReceivedFrom)
+				if err == nil && !app.UnsafeNoTrustIP {
+					app.P2p.Logger.Errorf("failed to connect to peer %s that just sent us an already-validated pubsub message, dropping it", peer.IDB58Encode(msg.ReceivedFrom))
```

Comment:
I don't understand what's going on here. Does it try to establish an outgoing connection to anyone that it gets a gossip message from?

Reply:
yeah, I'm trying to observe the IP to associate for trust scoring.

Reply:
in theory the connection pooling should be such that no new connection is necessary.